back/src/ML/app.js

```diff
@@ -61,6 +61,21 @@ function post(req,res,constants,request,client) {
 		}		
 	}
 
+	else if(req.params.type == 'surprise') {
+		if(req.params.what == 'update') {
+			if(!req && !req.body && !req.body.data) {
```

Comment:
use !req?.body?.data if you can.

Reply:
thats ts
apart from that I have made the changes